woocommerce/woocommerce
# Release tests with plugins installed

### All Submissions:


### Changes proposed in this Pull Request:

Closes:

- https://github.com/woocommerce/woocommerce-quality/issues/340
- https://github.com/woocommerce/woocommerce-quality/issues/341

The main change in this pull request is the adding of a job to test WC Core with other plugins installed. They are the [same plugins](https://github.com/woocommerce/woocommerce/blob/4a19efc739ec28127c80471dcaab035c2a024f09/.github/workflows/smoke-test-release.yml#L126-L138) from the Puppeteer release tests.

Other changes include:

1. Using composite actions located in `.github/actions/tests` to shorten the code in the `Smoke test release` workflow. Prior to this PR, the workflow already reached close to 700 lines of code without the plugin tests. In this PR, the workflow now only reaches around 660 lines of code but with the plugin tests already included.
1. Speaking of the composite actions mentioned above :point_up:, a few changes were made to them to add new useful [input parameters](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#inputs), and to simplify their steps further.
1. Artifact names are now readable. Instead of artifact names like `e2e-wp-latest--run-100`, they now look like `E2E test on release smoke test site with WP Latest (run 100)`.
1. The "Test against PHP" and "Test against WP Latest-X" job matrices no longer fail-fast.

### How to test the changes in this Pull Request:


1. Go to [Smoke test release](https://github.com/woocommerce/woocommerce/actions/workflows/smoke-test-release.yml) page.
2. In the "Run workflow" drop-down, select this branch (`e2e/release-plugins-0`).
3. In the "WooCommerce Release Tag", enter a release version that's not yet listed in the [WooCommerce Test Reports Dashboard > Releases](https://woocommerce.github.io/woocommerce-test-reports/release/) page.
4. Click "Run workflow".
5. Wait for the workflow to finish. If you specified an older release version, you might see failures in the E2E tests and that's because those versions might no longer be compatible with the current E2E tests.
6. When the workflow is done, head over to [WooCommerce Test Reports Dashboard > Releases](https://woocommerce.github.io/woocommerce-test-reports/release/). You should see your chosen release version, and the E2E results of certain plugins under it. API tests will be done on a future PR.
   <img width="440" alt="w9yXms6elq" src="https://user-images.githubusercontent.com/4509348/227176578-522cdf42-534d-477b-856f-79182093b0bd.png">

### Other information:


### FOR PR REVIEWER ONLY:
